### PR TITLE
feat: poetry port

### PR DESCRIPTION
**Pull request by luckasRanarison**
Solve [MET-604](https://linear.app/metatypedev/issue/MET-604/ghjk-poetry-plugin-up)

#### Migration notes

...

- [ ] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

File: ports/pipi.ts [hunk 1]
```diff
@@ -31,11 +31,15 @@ export const manifest = {
...
+  dependencies: zod.array(zod.object({
+    name: zod.string(),
+    version: zod.string().nullish(),
+  })).nullish(),
 }).passthrough();
 
 export type PipiInstallConf =
   & InstallConfigSimple
-  & zod.input<typeof confValidator>;
+  & Omit<zod.infer<typeof confValidator>, "dependencies">; // FIXME: Make it public?
```
Comment (Yohe-Am) [inline ports/pipi.ts:42]: Yeah, feel free to make it public. Rename the field to `peerDeps`. Pip install does handle normal deps actually.

File: ports/pipi.ts [hunk 2]
```diff
@@ -98,9 +102,16 @@ export class Port extends PortBase {
...
+
+    const dependencies = conf.dependencies?.map((dep) =>
+      dep.version ? [dep.name, dep.version].join("==") : dep.name
+    );
+
     await $`${
       depExecShimPath(std_ports.cpy_bs_ghrel, "python3", args.depArts)
-    } -m pip -qq install ${conf.packageName}==${args.installVersion}`
+    } -m pip -qq install ${conf.packageName}==${args.installVersion} ${
+      dependencies ?? []
```
Comment (Yohe-Am) [inline ports/pipi.ts:113]: Shouldn't this be adhering to the pkgName==pkgVersion format?

**Review by Yohe-Am — APPROVED** [review]

File: ports/pipi.ts (hunk 2 above)
Comment (luckasRanarison) [reply ports/pipi.ts]: Isn't it already the case?
https://github.com/luckasRanarison/ghjk/blob/met-604-ghjk-poetry-plugin-up/ports%2Fpipi.ts#L106-L108
Or what do you exactly mean?

**Review by Yohe-Am — APPROVED** [review]

**Comment by Yohe-Am** [comment]
@luckasRanarison what's the status on these two PRs?

**Comment by luckasRanarison** [comment]
@Yohe-Am sorry for the late reply. Do you have other change requests for this one? Otherwise, It's already working. And about the other one, I was lost in trying to inject `buildDepConfigs` in `ListAllargs`, I'm trying to figure that out now...

**Pull request merged**